kotlin/android/src/main/java/com/reactnativesecurekeystore/SecureKeystoreImpl.kt

```diff
@@ -215,21 +226,91 @@ class SecureKeystoreImpl(
         }
     }
 
-    override fun retrieveGenericKey(account: String): List<String> {
-        val privateKey= preferences.getPreference(Util.getPrivateKeyId(account), "")
-        val publicKey= preferences.getPreference(Util.getPublicKeyId(account), "")
-        val keyPair=ArrayList<String>()
-        keyPair.add(privateKey)
-        keyPair.add(publicKey)
-        return keyPair
+    override fun retrieveGenericKey(account: String, context: Any): List<String> {
+        try {
+            val privateKeyAlias = Util.getPrivateKeyId(account)
+            val publicKeyAlias = Util.getPublicKeyId(account)
+
+            val keyPair = ArrayList<String>()
+
+            val fragmentActivity = context as? FragmentActivity
```

Comment:
shouldn't we use Activity instead of FragmentActivity?
By casting it as FragmentActivity, we are restricting secure-keystore to not get used by simple activities.

Reply:
Casting to FragmentActivity is required as the `BiometricPrompt()` accepts only Fragments or fragmentActivity as parameters